pulsar-client-cpp/lib/MessageId.cc

```diff
@@ -79,7 +83,12 @@ MessageId MessageId::deserialize(const std::string& serializedMessageId) {
     if (!idData.ParseFromString(serializedMessageId)) {
         throw std::invalid_argument("Failed to parse serialized message id");
     }
-
+    if (idData.has_first_chunk_message_id()) {
+        auto firData = idData.first_chunk_message_id();
+        return MessageId(
+            MessageId(firData.partition(), firData.ledgerid(), firData.entryid(), firData.batch_index()),
+            MessageId(idData.partition(), idData.ledgerid(), idData.entryid(), idData.batch_index()));
+    }
     return MessageId(idData.partition(), idData.ledgerid(), idData.entryid(), idData.batch_index());
```

Comment:
We have `toMessageId` helper method in `MessageIdUtil.h`, I think we can simplify the code to

```c++
    if (idData.has_first_chunk_message_id()) {
        return {toMessageId(idData.first_chunk_message_id()), toMessageId(idData)};
    }
    return toMessageId(idData);
```